### ISSUE TITLE
Support JSON output

### ISSUE THREAD

**Issue by moshekaplan** [moshekaplan/palo_alto_firewall_analyzer#54]
Support generating JSON output, instead of only text-based.

**Comment by moshekaplan** [comment]
> It would be interesting to add the quantity of analyzed elements to create custom KPIs. Did you have any specific method in mind to accomplish this?

Nope, although if I'm understanding correctly, it would seem that those sorts of aggregate queries would best be moved into a separate 'reporting' module, so that reporting (and each report) would be independent of everything else.

**Comment by Nawtest** [comment]
No, what I want is to extract the totals from the analyzed information so that every person using the tool can utilize them to implement the necessary KPIs. Not calculate the KPIs within the tool itself.

**Comment by moshekaplan** [comment]
Could you give any examples of what totals you'd want to extract?

**Comment by Nawtest** [comment]
I was thinking of something like this:

![image](https://github.com/moshekaplan/palo_alto_firewall_analyzer/assets/26645553/4af5b332-a149-4d4d-86f0-d2c332678df6)

**Comment by Nawtest** [comment]
And in txt file:
![image](https://github.com/moshekaplan/palo_alto_firewall_analyzer/assets/26645553/23bacace-ba56-4548-a0dd-1338618ec6ce)

**Comment by Nawtest** [comment]
I have been modifying the JSON output to be able to exploit the data a bit more, I'll show you an image.

![image](https://github.com/moshekaplan/palo_alto_firewall_analyzer/assets/26645553/58cefb73-e90e-418d-8d52-6e474a70d296)

The idea is to add another parameter "--detail" to break down the text info.

**Comment by moshekaplan** [comment]
Apologies for the delayed response.

I like the idea, but my only concern is that it's not immediately clear what the fractions represent. Especially as some policies or objects are complex enough that a single object could have multiple failures. (e.g., a rule with a non-resolving FQDN can have entries in the source, destination, or both)

**Comment by Nawtest** [comment]
Don't worry, yes, I understand your concern. For now, I'm handling the simple ones. When I get to those types of rules, I guess I'll have to restructure the detail output. Let's see how it goes...
